ext/mysqli/mysqli.c

```diff
@@ -988,19 +988,22 @@ PHP_METHOD(mysqli_stmt, __construct)
 	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|s!", &mysql_link, mysqli_link_class_entry, &statement, &statement_len) == FAILURE) {
 		RETURN_THROWS();
 	}
-
 	MYSQLI_FETCH_RESOURCE_CONN(mysql, mysql_link, MYSQLI_STATUS_VALID);
 
 	stmt = (MY_STMT *) ecalloc(1, sizeof(MY_STMT));
-	stmt->stmt = mysql_stmt_init(mysql->mysql);
-	if (stmt->stmt && statement) {
-		mysql_stmt_prepare(stmt->stmt, (char *)statement, statement_len);
-	}
 
-	if (!stmt->stmt) {
+	if (!(stmt->stmt = mysql_stmt_init(mysql->mysql))) {
 		efree(stmt);
 		RETURN_FALSE;
 	}
+
+	if (statement) {
+		if (mysql_stmt_prepare(stmt->stmt, statement, statement_len)) {
+			MYSQLI_REPORT_STMT_ERROR(stmt->stmt);
```

Comment:
Needs an efree(stmt) afterwards?

Reply:
I don't understand why. I copied this from `mysqli_stmt_prepare` and there is no `efree` there. Besides I don't think that method allocates memory that I would need to deallocate myself. Can you explain it to me please?

Reply:
Because this is the funcion that allocates stmt (line 993). It will only be placed inside a resource below.

Reply:
Now I get it. I think I fixed it but can you please check if it is correct now?